Core/src/RequestHandler.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * Copyright 2023 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Core;
+
+use Google\ApiCore\Serializer;
+
+class RequestHandler
+{
+    use EmulatorTrait;
+    use RequestCallerTrait;
+
+    /**
+     * @var Serializer
+     */
+    private $serializer;
+
+    private $gapics;
+
+    /**
+     * @param array $config
+     */
+    public function __construct(
+        Serializer $serializer,
+        array $gapicClasses,
+        array $clientConfig = []
+    ) {
+        //@codeCoverageIgnoreStart
+        $this->serializer = $serializer;
+        $clientConfig['serializer'] = $serializer;
+
+        $this->setRequestWrapper(new GapicRequestWrapper($clientConfig));
+
+        // Adds some defaults
+        // gccl needs to be present for handwritten clients
+        $clientConfig += [
+            'libName' => 'gccl',
+            'transport' => $this->getDefaultTransport(),
+            'emulatorHost' => null
+        ];
+
+        if ((bool) $clientConfig['emulatorHost']) {
+            $emulatorConfig = $this->emulatorGapicConfig($clientConfig['emulatorHost']);
+            $clientConfig = array_merge(
+                $clientConfig,
+                $emulatorConfig
+            );
+        }
+        //@codeCoverageIgnoreEnd
+        
+        // Initialize the Gapic classes and store them in memory
+        $this->gapics = [];
+        foreach ($gapicClasses as $cls => $obj) {
```

Comment:
The GAPICs are instantiated here and stored as a map value in the `RequestHandler` itself. This is done so that we can reuse the same GAPIC object for multiple operations and we avoid a class instantiation every time we need to call a GAPIC method.